internal/middleware/compression.go

```diff
@@ -15,12 +15,59 @@
 package middleware
 
 import (
+	"bytes"
 	"compress/gzip"
 	"compress/zlib"
 	"io"
 	"net/http"
 )
 
+const (
+	headerContentEncoding = "Content-Encoding"
+	headerValueGZIP       = "gzip"
+)
+
+type CompressRoundTripper struct {
+	http.RoundTripper
+}
+
+func NewCompressRoundTripper(rt http.RoundTripper) *CompressRoundTripper {
+	return &CompressRoundTripper{
+		RoundTripper: rt,
+	}
+}
+
+func (r *CompressRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+
+	gzipWriter := gzip.NewWriter(nil)
+
+	// Gzip the body.
+	buf := bytes.NewBuffer([]byte{})
+	gzipWriter.Reset(buf)
```

Comment:
```golang
buf := bytes.NewBuffer([]byte{}) // Maybe preallocate this with WriteBufferSize? Just a TODO :)
gzipWriter := gzip.NewWriter(buf)
```

Reply:
Not sure, may be premature optimization. I suggest to keep as is and can be changed if profiling shows it is needed.

Reply:
The main suggestion not the extra comment is not an optimization, but more of a style change. Initialization of the zipper with the buffer instead of init with nil then reset

Reply:
Ah, I did not realize what you mean. Done.